### PR TITLE
fix(#923): Handle null values on Avro

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes https://github.com/kestra-io/kestra/issues/923

### CODE REVIEW

File: src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java [hunk 1]
```diff
@@ -113,18 +113,27 @@ private static Object objectDeserializer(Object value, Schema schema) {
...
-                    return AvroDeserializer.objectDeserializer(value, type);
-                } catch (Exception e) {
-                    return  false;
-                }
-            })
-            .findFirst()
-            .orElseThrow();
+        // first, if value is null, check if the null type exist to avoid generating a NPE
+        if(value == null) {
+            if(schema.getTypes().stream().map(s -> s.getType()).anyMatch(t -> Type.NULL == (t))) {
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java:118]:
```suggestion
            if(schema.getTypes().stream().filter(s -> s.getType() == Type.NULL == (t)).count() > 0) {
```
seems to be more readable no? I need to read the line twice

File: src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java [hunk 2]
```diff
@@ -113,18 +113,27 @@ private static Object objectDeserializer(Object value, Schema schema) {
...
+                return null;
+            }
+            else {
+                throw new NullPointerException("value is null but the schema type is not nullable: " + schema);
+            }
+        }
+
+        // then, evaluate each type and return the first that didn't generate an exception
+        for(var type  : schema.getTypes()) {
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java:127]: you are retesting the null value here?

Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java:127]: Yes, right, it should not be an issue but ideally we should remove it from the list of types to try.

File: src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java (hunk 1 above)
Comment (tchiotludo) [reply src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java:118]:
```suggestion
            if (schema.getTypes().stream().anyMatch(t -> t.getType() == Type.NULL)) {
```
ok this one is more readable IMO

File: src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java (hunk 2 above)
Comment (tchiotludo) [reply src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java:127]: ok don't worry, the list is really small

**Review by tchiotludo — APPROVED** [review]

File: src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java (hunk 1 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java:118]: Agreed

**Pull request merged**